src/components/badges/List.tsx

```diff
@@ -21,32 +20,23 @@ import walletStore from 'stores/WalletStore'
 function BadgeListSuspended() {
   const { account, walletsToNotifiedOfBeingDoxxed } = useSnapshot(walletStore)
   const { proofsCompleted } = useSnapshot(proofStore)
-  const {
-    emailDerivativeContracts = [],
-    ERC721derivativeContracts = [],
-    externalERC721derivativeContracts = [],
-  } = useSnapshot(SealCredStore)
-  const contractsOwned = useContractsOwned(GoerliContractsStore)
+  const { derivativeContracts } = useSnapshot(SealCredStore)
 
-  const ownedEmailDerivativeContracts = emailDerivativeContracts.filter(
-    (contractAddress) => contractsOwned.includes(contractAddress)
-  )
+  const contractsOwned = useContractsOwned(BadgesContractsStore)
 
-  const ownedExternalERC721DerivativeContracts =
-    externalERC721derivativeContracts.filter((contractAddress) =>
+  const ownedContractsByKey = dataShapeObject((ledgerName: string) =>
+    derivativeContracts[ledgerName].filter((contractAddress) =>
       contractsOwned.includes(contractAddress)
     )
-  const ownedERC721DerivativeContracts = ERC721derivativeContracts.filter(
-    (contractAddress) => contractsOwned.includes(contractAddress)
   )
 
   const proofsAvailableToMint = useProofsAvailableToMint()
 
   const hasProofs = proofsAvailableToMint.length
-  const hasDerivatives =
-    ownedExternalERC721DerivativeContracts.length +
-    ownedEmailDerivativeContracts.length +
-    ownedERC721DerivativeContracts.length
+  const hasDerivatives = Object.values(ownedContractsByKey).reduce(
```

Comment:
Maybe we can have a function to explain what is going on here?

Reply:
I simplified a little